pkg/engine/artifacts.go

```diff
@@ -30,111 +30,115 @@ func kubernetesContainerAddonSettingsInit(p *api.Properties) map[string]kubernet
+			isEnabled:       k.IsAddonEnabled(KeyVaultFlexVolumeAddonName),
 		},
 		DashboardAddonName: {
 			sourceFile:      "kubernetesmasteraddons-kubernetes-dashboard-deployment.yaml",
 			base64Data:      k.GetAddonScript(DashboardAddonName),
 			destinationFile: "kubernetes-dashboard-deployment.yaml",
-			isEnabled:       k.IsDashboardEnabled(),
+			isEnabled:       k.IsAddonEnabled(DashboardAddonName),
 		},
 		ReschedulerAddonName: {
 			sourceFile:      "kubernetesmasteraddons-kube-rescheduler-deployment.yaml",
 			base64Data:      k.GetAddonScript(ReschedulerAddonName),
 			destinationFile: "kube-rescheduler-deployment.yaml",
-			isEnabled:       !p.IsAzureStackCloud() && k.IsReschedulerEnabled(),
+			isEnabled:       k.IsAddonEnabled(ReschedulerAddonName),
 		},
 		NVIDIADevicePluginAddonName: {
 			sourceFile:      "kubernetesmasteraddons-nvidia-device-plugin-daemonset.yaml",
 			base64Data:      k.GetAddonScript(NVIDIADevicePluginAddonName),
 			destinationFile: "nvidia-device-plugin.yaml",
-			isEnabled:       !p.IsAzureStackCloud() && p.IsNvidiaDevicePluginCapable() && p.IsNVIDIADevicePluginEnabled(),
+			isEnabled:       k.IsAddonEnabled(NVIDIADevicePluginAddonName),
 		},
 		ContainerMonitoringAddonName: {
 			sourceFile:      "kubernetesmasteraddons-omsagent-daemonset.yaml",
 			base64Data:      k.GetAddonScript(ContainerMonitoringAddonName),
 			destinationFile: "omsagent-daemonset.yaml",
-			isEnabled:       !p.IsAzureStackCloud() && k.IsContainerMonitoringEnabled(),
+			isEnabled:       k.IsAddonEnabled(ContainerMonitoringAddonName),
 		},
 		IPMASQAgentAddonName: {
 			sourceFile:      "ip-masq-agent.yaml",
 			base64Data:      k.GetAddonScript(IPMASQAgentAddonName),
 			destinationFile: "ip-masq-agent.yaml",
-			isEnabled:       k.IsIPMasqAgentEnabled(),
+			isEnabled:       k.IsAddonEnabled(IPMASQAgentAddonName),
 		},
 		AzureCNINetworkMonitorAddonName: {
 			sourceFile:      "azure-cni-networkmonitor.yaml",
 			base64Data:      k.GetAddonScript(AzureCNINetworkMonitorAddonName),
 			destinationFile: "azure-cni-networkmonitor.yaml",
-			isEnabled:       o.IsAzureCNI() && k.IsAzureCNIMonitoringEnabled(),
+			isEnabled:       k.IsAddonEnabled(AzureCNINetworkMonitorAddonName),
 		},
 		DNSAutoscalerAddonName: {
 			sourceFile:      "dns-autoscaler.yaml",
 			base64Data:      k.GetAddonScript(DNSAutoscalerAddonName),
 			destinationFile: "dns-autoscaler.yaml",
-			// TODO enable this when it has been smoke tested
-			//common.IsKubernetesVersionGe(p.OrchestratorProfile.OrchestratorVersion, "1.12.0"),
-			isEnabled: false,
+			isEnabled:       k.IsAddonEnabled(DNSAutoscalerAddonName),
 		},
 		CalicoAddonName: {
 			sourceFile:      "kubernetesmasteraddons-calico-daemonset.yaml",
 			base64Data:      k.GetAddonScript(CalicoAddonName),
 			destinationFile: "calico-daemonset.yaml",
-			isEnabled:       k.NetworkPolicy == NetworkPolicyCalico,
+			isEnabled:       k.IsAddonEnabled(CalicoAddonName),
+		},
+		AzureNetworkPolicyAddonName: {
```

Comment:
Moved Azure NetworkPolicy addon into the user-configurable space as it's more appropriate to be there.